City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-4529: Vare views filters

# Daycare search filters [UHF-4529](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4529)
Added "provided languages" and "daycare options" filters to the daycare search view.

## What was done
* Added Kasko specific daycare options views filter.
* Added provided languages views filter for TPR Units.
* Added views relationship between TPR Unit and Ontology word details.
* Enabled better_exposed_filters module for Kasko.
* Added the new filters to Kasko's daycare search.

## How to install
* Make sure your Kasko instance is up and running on the `UHF-4529-vare-views-filters` branch.
* Update the TPR module:
  * `composer require drupal/helfi_tpr:dev-UHF-4529-views-filters`
* Update the Platform config module:
  * `composer require drupal/helfi_platform_config:dev-UHF-4529-vare-translations`
* Run:
  * `make drush-cim drush-cr`
  * `make shell`
  * `drush locale:check && drush locale:update && drush cr`
* With old data, it's probably a good idea to update the units and ontology word details also. Run also:
  * `drush migrate:import tpr_unit`
  * `drush migrate:import tpr_ontology_word_details`

## How to test
* Go to a page containing the daycare search.
* Make sure you have multiple daycare units visible and published.
  * You could add "Daycare Susanna" to test with Northern Sami.
* You should see two new filters:
  * The language filter filters daycares using the unit's provided languages. It's possible to check multiple languages with checkboxes.
  * The option filter filters daycares according to their service types.
* Check that you can filter the results using the new filters.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/111
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/301
